modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,344 @@
+namespace face{
+const std::string face_cascade_name = "lbpcascade_frontalface_improved.xml";
+const std::string meanshapefile ="opencv2/modules/face/data/mean_shape.txt";
+class ShapePredictor : public cv::face::FacemarkKazemi{
+public:
+    virtual int getNumLandmarks() const {return numlandmarks_;}
+    virtual int getNumFaces() const {return numfaces_;}
+    ShapePredictor(){
+        //get name  of file containing meanshape
+        ifstream f(meanshapefile.c_str());
+        string s;
+        stringstream ss;
+        //read value of annotations
+        while(getline(f,s)){
+            unsigned long i=0;
+            char x1[100],y1[100];
+            while(s[i]!=','){
+                x1[i]=s[i];
+                i++;
+            }
+            string x,y;
+            x=string(x1);
+            Point2f pt;
+            ss<<x;
+            ss>>pt.x;
+            ss.clear();
+            i++;
+            int j=0;
+            while(i<s.length()){
+                y1[j++]=s[i];
+                i++;
+            }
+            y=string(y1);
+            ss<<y;
+            ss>>pt.y;
+            ss.clear();
+            //initialise the meanshape
+            meanshape_.push_back(pt);
+        }
+        //initialise other variables
+        numfaces_=1;
+        numlandmarks_=194;
+    }
+    virtual ~ShapePredictor(){}
+
+    /*Reads the landmarks from the txt file.
+    Each file's first line should give the path of the image whose
+    landmarks are being described in the file.Then in the subsequent
+    lines there should be coordinates of the landmarks in the image
+    i.e each line should be of the form "x,y"(ignoring the double quotes)
+    where x represents thee x coordinate of the landmark and y represents
+    the y coordinate of thee landmark.*/
+    bool getData(std::vector<string> filename,std::vector<std::vector<cv::Point2f> > &_trainlandmarks
+                ,std::vector<string> &_trainimages);
+    /*This function gets tthe relative shape of the face scaled and centred
+      according to the bounding rectangleof the detected face*/
+    bool getMeanShapeRelative(std::vector<cv::Rect> face,std::vector<std::vector<cv::Point2f> > & initialshape_);
+    /*@param faces - stores the bounding boxes of the face whose initial shape has
+    to be found out*/
+    /*This function calculates mean shape while training.
+    Only called when new training data is supplied by the train function.*/
+    bool calcMeanShape(std::vector< std::vector<cv::Point2f> > &_trainlandmarks,InputArrayOfArrays _trainimages);
+
+    /*This functions scales the annotations according to a common size which is considered for all images*/
+    bool scaleData(std::vector< std::vector<cv::Point2f> > & _trainlandmarks,
+                                vector<Mat> & _trainimages ,Size s=Size(200,200) );
+    //function to get bounding rectangle
+    bool getBoundingRect(cv::Mat src,std::vector<cv::Rect> &facep);
+private:
+    int numlandmarks_;
```

Comment:
For the next, and adapt the current. There is this code style, where internal function variables have the same name as the parameters passed to the function declaration. Then people tend to add a _ in front of it to avoid conflicts. But in my opinion; thats a bad practice...